staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -121,6 +132,197 @@ const (
+	//Policy specifies the admission control policy definition which the Param used on
+	// Required.
+	Policy string `json:"policy,omitempty" protobuf:"bytes,1,rep,name=policy"`
+
+	//Param specifies the parameter resource used to configure the admission control policy
+	// +optional
+	Param string `json:"param,omitempty" protobuf:"bytes,2,rep,name=param"`
+
+	//MatchResources describe a list of match criteria which should match for the admission control policy to validate on.
+	// +optional
+	MatchResources MatchResources `json:"matchResources,omitempty" protobuf:"bytes,3,rep,name=matchResources"`
+}
+
+// MatchResources decides whether to run the admission control policy on an object based
+// on whether meet the match criteria.
+type MatchResources struct {
+	// Namespaces specifies the namespaces which the admission control policy should validate on.
+	// +optional
+	Namespace []string `json:"namespace,omitempty" protobuf:"bytes,1,opt,name=namespace"`
+	// ExcludeNamespace specifies the namespaces which the validating admission policy should not validate on.
+	// +optional
+	ExcludeNamespace []string `json:"excludeNamespace,omitempty" protobuf:"bytes,2,opt,name=excludeNamespace"`
+	// NamespaceSelector decides whether to run the admission control policy on an object based
+	// on whether the namespace for that object matches the selector. If the
+	// object itself is a namespace, the matching is performed on
+	// object.metadata.labels. If the object is another cluster scoped resource,
+	// it never skips the admission control policy.
+	//
+	// For example, to run the admission control policy on any objects whose namespace is not
+	// associated with "runlevel" of "0" or "1";  you will set the selector as
+	// follows:
+	// "namespaceSelector": {
+	//   "matchExpressions": [
+	//     {
+	//       "key": "runlevel",
+	//       "operator": "NotIn",
+	//       "values": [
+	//         "0",
+	//         "1"
+	//       ]
+	//     }
+	//   ]
+	// }
+	//
+	// If instead you want to only run the admission control policy on any objects whose
+	// namespace is associated with the "environment" of "prod" or "staging";
+	// you will set the selector as follows:
+	// "namespaceSelector": {
+	//   "matchExpressions": [
+	//     {
+	//       "key": "environment",
+	//       "operator": "In",
+	//       "values": [
+	//         "prod",
+	//         "staging"
+	//       ]
+	//     }
+	//   ]
+	// }
+	//
+	// See
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
+	// for more examples of label selectors.
+	//
+	// Default to the empty LabelSelector, which matches everything.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty" protobuf:"bytes,3,opt,name=namespaceSelector"`
+	// LabelSelector decides the match criteria on resource based on its labels.
+	// +optional
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty" protobuf:"bytes,4,opt,name=labelSelector"`
```

Comment:
Why fully explain label selectors on `NamespaceSelector` and say so little for `LabelSelector`?